bundler/lib/dependabot/bundler/update_checker/latest_version_finder/dependency_source.rb

```diff
@@ -0,0 +1,149 @@
+# frozen_string_literal: true
+
+module Dependabot
+  module Bundler
+    class UpdateChecker
+      class LatestVersionFinder
+        class DependencySource
+          require_relative "../shared_bundler_helpers"
+          include SharedBundlerHelpers
+
+          RUBYGEMS = "rubygems"
+          PRIVATE_REGISTRY = "private"
+          GIT = "git"
+          OTHER = "other"
+
+          attr_reader :dependency, :dependency_files, :repo_contents_path,
+                      :credentials
+
+          def initialize(dependency:,
+                         dependency_files:,
+                         repo_contents_path: nil,
+                         credentials:)
+            @dependency          = dependency
+            @dependency_files    = dependency_files
+            @repo_contents_path  = repo_contents_path
```

Comment:
✂️ `repo_contents_path` unused?

Reply:
I think it might be used in the SharedHelpers mixin methods, will double check

Reply:
It isn't it can go 👍🏻